### DISCUSSION TITLE
Is there anyway to remove the unwanted z-hop across prints when there are multiple instances?

### DISCUSSION THREAD

**Discussion by SupItsSaige** [supermerill/SuperSlicer#3335] [Q&A]
I'm printing 4 instances layer-by-layer. The default behavior is printing each layer in the same order every time meaning there is this z-hop across parts once the new layer is started. Other slicers I've used (Ideamaker and simplify3d) would start the next layer at the instance it finished the previous layer on. Is there a setting to force that behavior? The current behavior is causing lots of stringing only on one instance because of the pictured z-hop at the start of new layers.
![image](https://user-images.githubusercontent.com/115582798/195169154-3ffdf6c0-c5c0-4ca5-afa7-4d1ffe0d712c.png)

**Comment by neophyl** [comment]
Not as far as I'm aware no.  One of the posters on the Prusa forum uses extra sacrificial object(s) to somewhat control tool paths.  In this case if you added a small diameter cylinder of the same height then wouldn't the z-hop stringing be on that and not your primary objects ?

**Reply by SupItsSaige** [reply]
Thanks! I could see this being a good band aid in the moment, but for someone constantly printing multiple layer-by-layer instances it doesn't seem viable to have so many unnecessary throw away objects. Has this feature been requested yet? I've been searching through the requests, but haven't found any. I think I'll make a request if not.

**Reply by Garryb123** [reply]
The behavior you request is logical and the most efficient, not to mention effective at reducing stringing; it should be methodology used.

**Comment by vovodroid** [comment]
I would suggest "semi-sequential printing" - i.e. print each object for height between nozzle tip and extruder (5-10mm) and then move to next object. Thus we avoid collisions and save travels.

**Reply by vovodroid** [reply]
@SupItsSaige , I implemented "semi-parallel" printing, e.g. print layer range for each object (with some limitation though).
Would you like to test it?

**Comment by vovodroid** [comment]
See #3445

**Reply by SupItsSaige** [reply]
First of all I love that functionality! It was one of the few things I missed from Simpify3D and I'd really like to see something like this added. I would happily test it out!

**Reply by vovodroid** [reply]
> I would happily test it out!

Here is the branch - https://github.com/vovodroid2/SuperSlicer/tree/semi-parallel-printing
Could you build it?

**Reply by SupItsSaige** [reply]
I'm very sorry for the late replies but this end of the year has been crazy busy. I was looking at the branch linked and tried my best to download it. I am unsure how to build it, especially since it says windows version is prebuilt. I am a little unsure how to get it working. Sorry for being a struggle.

**Reply by vovodroid** [reply]
> I am unsure how to build it

Generally according to instructions: https://github.com/vovodroid2/SuperSlicer/blob/semi-parallel-printing/doc/How%20to%20build%20-%20Windows.md, but let's simplify them:

1) Download this branch as ZIP https://codeload.github.com/vovodroid2/SuperSlicer/zip/refs/heads/semi-parallel-printing
2) Install VS2019 Community from https://learn.microsoft.com/en-us/visualstudio/releases/2019/release-notes with C++ option checked.
3) From project directory run `.\build_win.bat -d "..\SuperSlicer-deps"  -p low -c Release -s all`
4) Run Superslicer from folder "SuperSlicer\build\src\Release"
5) Enjoy )))
